app/Tquery/Config/TqDictDef.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Tquery\Config;
+
+use App\Exceptions\FatalExceptionFactory;
+use App\Models\UuidEnum\DictionaryUuidEnum;
+
+readonly class TqDictDef
+{
+    public string $dictionaryId;
+
+    public function __construct(
+        public TqDataTypeEnum $dataType,
+        string|DictionaryUuidEnum $dictionary,
+    ) {
+        if (!$this->dataType->isDict()) {
+            FatalExceptionFactory::tquery()->throw();
+        }
+        $this->dictionaryId = is_string($dictionary) ? $dictionary : $dictionary->value;
```

Comment:
Ta logika się powtarza, może zrobić jakiegoś helpera?

Reply:
wydaje mi się, że jeszcze bym tego nie wyciągał, to jest powtórzone 2 razy, i nie jest jakąś wielką logiką. A IDE ogarnia sobie typy dobrze przy tym